Fix code folding icon color not matching defined color (3.x)

PhoenixFlame101 opened the pull request:
Fix #58379 for `3.x`

Cherrypick of #58457 from `master`

Caused by reusing icons from the main editor in the code editor. These
icons were converted based on the main editor theme and not the code
editor theme.

 - Create new icons for use specifically in the code editor
 - Add these icons to the exceptions when converting dark theme icons
   to light theme automatically
 - Change the default value of the code folding color to match previous
   color
 - Code folding icon is now pure white by default to correctly match
   the color defined in settings

<!--
Pull requests should always be made for the `master` branch first, as that's
where development happens and the source of all future stable release branches.

Relevant fixes are cherry-picked for stable branches as needed.

Do not create a pull request for stable branches unless the change is already
available in the `master` branch and it cannot be easily cherry-picked.
Alternatively, if the change is only relevant for that branch (e.g. rendering
fixes for the 3.2 branch).
-->

Calinou committed:
Add a per-light volumetric fog energy property

Per-light energy gives more control to the user on the final result of
volumetric fog. Specific lights can be fully excluded from volumetric fog
by setting their volumetric fog energy to 0, which improves performance
slightly. This can also be used to prevent short-lived dynamic effects
from poorly interacting with volumetric fog, as it's updated over several
frames by default unless temporal reprojection is disabled.

Volumetric fog shadows now obey Light3D's Shadow Opacity property as well.

The shadow fog fade property was removed as it had little visible impact
on the final scene's rendering.

SaracenOne committed:
Importer option to apply the root basis for meshes/skins/animations

Calinou committed:
Add Environment properties to control fog rendering on background sky

Values lower than 1.0 can be used to make the fog rendering not fully
obstruct the sky. This can be desired when using fog as a purely
atmospheric effect, without intending to use fog for open world fog
fading.

When set to 0.0, fog rendering behavior will be similar to Godot 3.x
where sky rendering was never affected by fog.

Calinou committed:
Rename `or_lesser` range property hint to `or_less`

"less" should be used for quantity, rather than "lesser".

Existing scripts that use `or_lesser` in `_get_property_list()`
will need to be updated to account for this change.

Calinou committed:
Improve the editor's Add Metadata dialog

- Mention the node name in the dialog title.
- Improve error messages to be more descriptive.

piiertho committed:
enhancement: Rename declared property AnimationNodeTransition::input_count to AnimationNodeTransition::enabled_inputs

twaritwaikar committed:
VCS: Port Godot 3.5's VCS features to GDExtension

fire-forge committed:
Improvements to multi-node editing
- Show revert button for properties that are not default for all selected nodes
- Show property documentation tooltips
- Show common class name and icon and number of selected nodes in EditorPath, e.g. "Node2D (4 Selected)"
- Hide metadata for MultiNodeEdit and AnimationMultiTrackKeyEdit
- Hide script for MultiNodeEdit

Xwdit committed:
Some improvements to the SceneTreeTimer

Calinou committed:
Rename Curve/Curve2D/Curve3D/Gradient `interpolate()` to `sample()`

"sampling" is a more accurate term than "interpolating" for what's
happening when using that function.

Calinou committed:
Improve null and object printing to avoid confusion with arrays

- Use different syntax for object printing to avoid confusion with arrays.
- Print null as `<null>` to avoid confusion with a string `"null"`.
- Display `<empty>` in editor resource pickers to avoid confusion
  with array-based properties.

twaritwaikar committed:
VCS: Load a GDExtension instead of Script + UI port from 3.5

twaritwaikar committed:
VCS: Fix empty metadata items being moved across stages

twaritwaikar committed:
VCS: Replace usage of varray() by using callable_mp::bind()

clayjohn committed:
Implement Physical Light Units as an optional setting.

This allows light sources to be specified in physical light units in addition to the regular energy multiplier. In order to avoid loss of precision at high values, brightness values are premultiplied by an exposure normalization value.

In support of Physical Light Units this PR also renames CameraEffects to CameraAttributes.

BastiaanOlij committed:
Extracting render buffers and changing it to a more generic solution

A-Lamia committed:
fix: add uri_decode to root_uri #63388

twaritwaikar committed:
VCS: Improve VCS UI/UX by QoL changes

The editor will now use the project path i.e. the place where the root of
the repo is supposed to be according to the user. This project path is
also sent into the plugin and so out-of-directory asset folders can also be
maintained this way.

asmaloney committed:
[doc] Use f-strings throughout make_rst.py

Based on discussion in #64230

KoBeWi committed:
Rework oriented containers

maximkulkin committed:
Add GraphNode 'selected' and 'deselected' signals, simplify GraphEdit

Tim-Fronsee committed:
Fix TextEdit::gutters_width (total gutter width) when adding & removing a gutter by calling TextEdit::_update_gutter_width in TextEdit::add_gutter & TextEdit::remove_gutter

Update TextEdit gutters subcase, gutter add and remove to ensure gutter total width is correct

Fix test_code_edit symbol lookup test case to include padding (+2)

Chaosus committed:
Rename `uniform` to `parameter` across the engine

bruvzg committed:
Add support for system dark mode and accent color detection (macOS and Windows). Add support for dark mode title bar on Windows.

yedpodtrzitko committed:
fix: graceful exit on macOS missing Vulkan

groud committed:
Improve SplitContainer behavior, keeping asked split whenever possible

Faless committed:
[Web] Require threads, rtti, allow optimize=speed.

Update export names (web[_dlink]_[release|debug].zip).

The Build with dynamic linking is broken due to high number of imports
in output wasm (likely emscripten regression issue 15487).

Atlinx committed:
Fix inferred GDScript enum values not appearing in inspector

"enum_values" originally wasn't being forwarded to the new type inside "reduce_identifier_from_base", which caused hint strings derived from the new type to be blank, which ultimately caused an empty enum dropdown menu.

bruvzg committed:
[TextServer] Add support for trimming edge spaces on line break.

MatthewZelriche committed:
Fix dropped XEvents early in main window lifetime.

The DisplayServerX11 constructor processes pending events shortly
after constructing the main window. However, it discards pending events
on the event queue that it is not interested in. This results in these
events never making it to the main events thread and as a result are
never processed. We need to save the events we don't handle in
DisplayServerX11 so that they can be resent for later handling by the
events thread.

Mickeon committed:
Mark Script button if it's tool in Scene Tree Editor

Also adds a note on the tooltip if the Script is tool.

The color is the same one used on the Script Editor's script list.

raulsntos committed:
Add support for system dark mode (Linux)

- Use `org.freedesktop.appearance color-scheme` to support system dark mode.

groud committed:
Fixes inherited scenes not working in TileSet scenes collection sources

akien-mga committed:
Merge pull request #65042 from YuriSizov/editor-docks-tabbar-bg

Commit:
Additional fixes to the Android `get_current_dir()` implementation.

akien-mga committed:
Merge pull request #64327 from KoBeWi/_update_callable

bruvzg committed:
Fix MSDF fonts color modulation.

akien-mga committed:
Merge pull request #64554 from nongvantinh/material_are_not_applied

akien-mga committed:
Merge pull request #47744 from KoBeWi/press_F_to_play_exit

akien-mga committed:
Merge pull request #59855 from KoBeWi/focus_thief

Chaosus committed:
Merge pull request #62717 from Chaosus/jps_pathfinding

akien-mga committed:
Merge pull request #65068 from BastiaanOlij/fix_dummy_mesh_allocation

akien-mga committed:
Merge pull request #63552 from RedMser/file-dialog-disambiguate

EditorFileDialog: disambiguate recent/favorite items

MatthewZelriche committed:
Fix minimize/maximize not taking effect in X11.

Attempts to construct an X11 window in an initial state of
minimized/maximized would fail due to the window being unmapped.
We simply check for failed mode changes during an unmap and reapply
them if necessary.

akien-mga committed:
Merge pull request #65087 from TokageItLab/change-skeleton-3d-api

Clean-up/change some Skeleton3D API

akien-mga committed:
Merge pull request #65089 from YuriSizov/editor-launch-bar-with-style

Improve style and add contextual highlight to the editor launch pad

akien-mga committed:
Merge pull request #64545 from asmaloney/doc-fstring-make_rst

[doc] Use f-strings throughout make_rst.py

MinusKube committed:
Prevent subwindow_focused from pointing to a window that is not a subwindow

akien-mga committed:
Merge pull request #65094 from Faless/web/4.x_templates_refactor

[Web] Require threads, rtti, allow optimize=speed.

akien-mga committed:
Merge pull request #65098 from bruvzg/msdf_modulate

Fix MSDF fonts color modulation.

YeldhamDev committed:
Fix switching `MenuBar`'s menus with the keys while mouse hovering

timothyqiu committed:
Fix crash when toggling flag property value

timothyqiu committed:
Remove unnecessary `print_line` in RendererDummy

zhehangd committed:
Basis constructor: row vectors -> column vectors

akien-mga committed:
Merge pull request #62157 from ChronicallySerious/front-port-vcs-plugin

VCS: Port Godot 3.5's VCS features to GDExtension

akien-mga committed:
Merge pull request #65121 from timothyqiu/print-line

Remove unnecessary `print_line` in RendererDummy

BastiaanOlij committed:
Fix vector type for signed tangent in mobile shader

akien-mga committed:
Merge pull request #63394 from Calinou/curve-gradient-rename-interpolate

Rename Curve/Curve2D/Curve3D/Gradient `interpolate()` to `sample()`

akien-mga committed:
Merge pull request #63413 from Calinou/volumetric-fog-add-per-light-energy-2

Add a per-light volumetric fog energy property

akien-mga committed:
Merge pull request #65095 from m4gr3d/update_get_current_dir_main

Additional fixes to the Android `get_current_dir()` implementation.

TokageItLab committed:
Discontinued spherical_interpolate_with in Transform3D

akien-mga committed:
Merge pull request #65130 from BastiaanOlij/fix_mobile_signed_tangent

Fix vector type for signed tangent in mobile shader

bruvzg committed:
[macOS] Handle accelerator and click events of the global menu items separately.

reduz committed:
Add support for scene/resource customization in export plugins

EditorExportPlugin adds a set of callbacks to allow customizing scenes, resources or subresources in all files exported:
* Can take scene files, resource files and subresources in all of them.
* Uses a cache for the converted files if nothing changes, so this work only happens if a file is modified.
* Uses hashing to differentiate export configuration caches.
* Removed the previous conversion code to binary, as this one uses existing stuff.

This API is useful in several scenarios:
* Needed by the "server" export platform to get rid of textures, meshes, audio, etc.
* Needed by text to binary converters.
* Needed by eventual optimizations such as shader precompiling on export, mesh merging and optimization, etc.

This is a draft, feedback is very welcome.

Mickeon committed:
Modify newlines in Editor Spin Slider tooltip

Makes both key suggestions easier to read.

akien-mga committed:
Merge pull request #65120 from timothyqiu/flags-skip

akien-mga committed:
Merge pull request #65092 from groud/inherited_scenes_tilesets

akien-mga committed:
Merge pull request #59919 from piiertho/enhancement/rename-AnimationNodeTransition-input_count-to_enabled_inputs

RedMser committed:
Fix #65122: disambiguate_filenames freeze

akien-mga committed:
Merge pull request #65131 from TokageItLab/transform-with

YuriSizov committed:
Add a lifecycle method for manual theme item caching to Control

akien-mga committed:
Merge pull request #65065 from Atlinx/fix/65010_enum-doesnt-show-up

akien-mga committed:
Merge pull request #64130 from A-Lamia/root_uri_decode

akien-mga committed:
Merge pull request #65142 from RedMser/fix-65122-freeze

Fix `EditorNode::disambiguate_filenames` freeze

akien-mga committed:
Merge pull request #63411 from Calinou/improve-null-object-dictionary-print

Improve null and object printing to avoid confusion with arrays

akien-mga committed:
Merge pull request #64247 from guilhermefelipecgs/fix_end_node

Re-add AnimationNodeStateMachine::end_node for root state machine

akien-mga committed:
Merge pull request #63893 from guilhermefelipecgs/fix_state_machine_is_playing

Fix AnimationNodeStateMachinePlayback::is_playing

KoBeWi committed:
Change Array arguments to TypedArray

paddy-exe committed:
Fix LinearDepth calculation for GLES3

This change will calculate the depth ndc differently for GLES3 and Vulkan as described in here:
stable:
https://docs.godotengine.org/en/stable/tutorials/shaders/advanced_postprocessing.html
godot 4 with vulkan:
https://docs.godotengine.org/en/latest/tutorials/shaders/advanced_postprocessing.html

m4gr3d committed:
Fix build failures for Android on Windows

The issue is caused by https://github.com/godotengine/godot/pull/64306 which makes use of a 3.7 feature while the current recommended python version is 3.5 for 3.x and 3.6 for master.

Chaosus committed:
Merge pull request #65160 from paddy-exe/fix-linear-depth-gles

YuriSizov committed:
Unite GradientEdit and GradientEditor as editor-only widget

MewPurPur committed:
Fixed a case for global function highlighting

neikeq committed:
C#: Fix Vector4 in godot_variant and missing marshaling

Vector4 and Vector4i were implemented incorrectly in godot_variant.
They were also missing their respective Variant conversion callbacks
(used for generic collections).

Took the chance to remove unnecessary native calls for creating
Variant from Vector4, as now it can be done from C# (which is faster).

raulsntos committed:
C#: Assume 64-bit types when type has no meta

When the C# bindings generator finds a type without meta assume the type
refers to the 64-bit version of the type:
- `float` is converted to `double`
- `int` is converted to `long`

clayjohn committed:
Merge pull request #65161 from m4gr3d/fix_android_build_failures_on_windows_main

Fix build failures for Android on Windows

neikeq committed:
Merge pull request #65164 from neikeq/csharp-fix-vector4-in-variant

C#: Fix Vector4 in godot_variant and missing marshaling

Faless committed:
Simplify network GDExtension bindings using EXBIND macros.

Simplify StreamPeer, PacketPeer, MultiplayerPeer extension.

Simplify and update WebRTC*Extension with newly supported types.

bruvzg committed:
[TextServer] Fix LCD AA fonts reading wrong glyphs.

fire-forge committed:
Various editor UI fixes (bottom panel corner radius and scene tab bar)
- Fix top corners of bottom panel not having rounded corners
- Fix scene tab bar background expanding vertically by corner radius
- Remove two unneccesary theme items from the editor theme:
	- EditorStyles/SceneTabFG
	- EditorStyles/SceneTabBG
- Remove an instance where the scene TabBar was having its styleboxes overriden to the default value, which has no effect.

fire-forge committed:
Decrease StyleBoxFlat corner detail in editor theme
Also makes some minor changes to other areas of the editor theme.

akien-mga committed:
Merge pull request #63751 from clayjohn/physical_light_units

Implement Physical Light Units in Vulkan Renderers

akien-mga committed:
Merge pull request #65073 from bruvzg/trim_spaces

[TextServer] Add support for trimming edge spaces on line break.

akien-mga committed:
Merge pull request #65175 from bruvzg/fix_lcd_gl_idx

[TextServer] Fix LCD AA fonts reading wrong glyphs.

akien-mga committed:
Merge pull request #64444 from timothyqiu/action-completion

Fix action name completion for `Input`

akien-mga committed:
Merge pull request #65162 from YuriSizov/editor-gradients-united

Unite `GradientEdit` and `GradientEditor` as editor-only widget

akien-mga committed:
Merge pull request #65107 from MatthewZelriche/X11MinimizeMaximizeFix

Fix minimize/maximize not taking effect in X11.

akien-mga committed:
Merge pull request #65108 from MinusKube/subwindow-focused-bug

Prevent subwindow_focused from pointing to a window that is not a subwindow

akien-mga committed:
Merge pull request #65117 from YeldhamDev/menubar_key_fix

Fix switching `MenuBar`'s menus with the keys while mouse hovering

akien-mga committed:
Merge pull request #65163 from MewPurPur/reuse-function-logic-for-global-funcs

Fix a case in global function highlighting

akien-mga committed:
Merge pull request #65124 from zhehangd/fix_basis

JohanAR committed:
Replace Vector2(i) with Size2(i) for methods returning a size

YuriSizov committed:
Add dumb and manual theme caching systems to Window

akien-mga committed:
Merge pull request #63901 from BastiaanOlij/render_buffers_reorg

akien-mga committed:
Merge pull request #65132 from bruvzg/global_menu_shortcuts_context

akien-mga committed:
Merge pull request #65088 from Mickeon/editor-scene-tree-tool-icon

BenniKane committed:
doc: Clarifying NavigationAgent and NavigationAgent2D velocity_computed requires avoidance_enabled to be true to emit

akien-mga committed:
Merge pull request #63968 from KoBeWi/finding_stuff_in_a_dictionary

akien-mga committed:
Merge pull request #64157 from Mickeon/remove-mysterious-is-playing

YuriSizov committed:
Fix theme propagation in various parts of the editor

akien-mga committed:
Merge pull request #64173 from smix8/navigation_debug_gridmap_edgeconnections_4.x

neikeq committed:
Merge pull request #65168 from raulsntos/dotnet/float64

C#: Assume 64-bit types when type has no meta

aaronfranke committed:
Fix a minor bug in the Projection binary decode logic

RedMser committed:
Allow searching for annotations in help

akien-mga committed:
Merge pull request #55617 from madmiraal/fix-55384

Geometror committed:
Refactor BitMap and add tests

Co-authored-by: Resul Çelik <resul_celik@hotmail.com>

akien-mga committed:
Merge pull request #65156 from YuriSizov/control-customizable-cache-p1

akien-mga committed:
Merge pull request #65192 from YuriSizov/control-customizable-cache-p2

Mickeon committed:
Fix Label & RichTextLabel's visible_ratio not working

bruvzg committed:
[TextServer] Fix excessive space trimming.

DissonantVoid committed:
fixed and optimized Area2/3D get_overlapping_bodies/areas

fixed a type and made it so area3d get_overlapping_x is similar to its counterpart func in area2d so that it uses TypedArray instead of Array and ERR_FAIL_COND_V_MSG instead of no message, also minimized array resize calls

akien-mga committed:
Merge pull request #64714 from Chaosus/string_remove_erase

akien-mga committed:
Merge pull request #64180 from aaronfranke/project-features-tools-only

akien-mga committed:
Merge pull request #65135 from reduz/export-customization-plugins

akien-mga committed:
Merge pull request #63479 from DarkKilauea/nav-link

akien-mga committed:
Merge pull request #64198 from Geometror/add-bitmap-tests

akien-mga committed:
Merge pull request #59602 from Calinou/editor-improve-add-metadata-dialog

akien-mga committed:
Merge pull request #65206 from aaronfranke/fix-projection-decode

akien-mga committed:
Merge pull request #65213 from Mickeon/fix-label-visible-ratio

akien-mga committed:
Merge pull request #65216 from bruvzg/trim_fix

akien-mga committed:
Merge pull request #65187 from JohanAR/vector_to_size

akien-mga committed:
Merge pull request #65212 from Faless/net/4.x_simplify_ext_bindings

akien-mga committed:
Merge pull request #60185 from Calinou/environment-fog-and-sky-affect

akien-mga committed:
Fix build after merge of #63479

jtnicholl committed:
Add `is_zero_approx` methods to `Vector2`, `3`, and `4`

Rindbee committed:
Fix getting an error message when repeatedly entering and exiting the tree

fire-forge committed:
Increase ItemList v_separation in editor theme
ItemList now uses the same value for v_separation as PopupMenu. This makes it easier to select items.

akien-mga committed:
Merge pull request #65223 from jtnicholl/is_zero_approx

akien-mga committed:
Merge pull request #64724 from KoBeWi/HVBoxContainer

akien-mga committed:
Merge pull request #65225 from Rindbee/fix-reparent-from-window-issue

akien-mga committed:
Merge pull request #65202 from BenniKane/patch-1

akien-mga committed:
SCons: Enforce version 4.0+ when `compiledb=yes` is used

Fixes #65233.

bruvzg committed:
[macOS] Simplify handling of menu accelerators, and send event to the focused window instead of main.

akien-mga committed:
Merge pull request #64952 from Chaosus/vs_rename_uniform_to_param

akien-mga committed:
Merge pull request #65236 from akien-mga/scons-compiledb-explicit-requirement

akien-mga committed:
Merge pull request #65210 from YuriSizov/control-customizable-cache-p3

akien-mga committed:
Merge pull request #65218 from xX-Void-Xx/area2d/3d-optimization

akien-mga committed:
Merge pull request #65026 from bruvzg/dark_mode

akien-mga committed:
Merge pull request #65242 from bruvzg/menu_event_multiwindow

akien-mga committed:
Merge pull request #65170 from KoBeWi/your_argument_is_TypedArray

akien-mga committed:
Merge pull request #64735 from KoBeWi/viewportshifter

Mickeon committed:
Improve Docs' inheritance tree icons on newline

Uses a Non-Breaking Space to prevent the icon from detaching from the name on newline, improving the look considerably.

YuriSizov committed:
Add ThemeOwner type for managing theme propagation and lookup

Chaosus committed:
Revert usage of typed array in `add_surface_from_arrays` parameters

YuriSizov committed:
Un-defer the initial theme changed notification

Co-authored-by: Rindbee <idleman@yeah.net>

AlmightyLaxz committed:
Fixed build failing when REAL_T double and dotnet enabled

Chaosus committed:
Merge pull request #65249 from Chaosus/revert_array_param

RedMser committed:
EditorTheme: Remove rounded corners for all popups

Expands #59045 to apply to all kinds of popups,
since they all render as separate windows.

AlmightyLaxz committed:
Add float arg to build_assemblies.py

Calinou committed:
Improve documentation related to audio input permissions

YuriSizov committed:
Fix property revert logic in the sectioned inspector

Co-authored-by: derammo <817160+derammo@users.noreply.github.com>

akien-mga committed:
Merge pull request #65261 from YuriSizov/property-MUST-revert

akien-mga committed:
Merge pull request #64917 from Tim-Fronsee/fix/add-gutter-total-width

MewPurPur committed:
Remove old syntax for custom class icon

raulsntos committed:
C#: Create script instance of reloaded scripts even if they're not tools

Scripts that are instantiated at some point will always be recreated
if they ever become placeholders to prevent non-tool scripts
instantiated manually by users to become placeholders, if they
do become placeholders due to errors that prevent instantiation
(such as a missing parameterless constructor) these scripts
will also be recreated replacing the temporary placeholder.

If a script is marked as a tool but becomes a non-tool script
in a rebuild, the script will become a placeholder and will
no longer be considered applicable to be replaced by an instance
since the user explicitly removed the Tool attribute.

akien-mga committed:
Merge pull request #65115 from raulsntos/linux/dark-mode

neikeq committed:
Merge pull request #65251 from AlmightyLaxz/fix-real_t-double-csharp-compiling

Fixed GodotSharp build failing when real_t is double

akien-mga committed:
Merge pull request #59589 from Calinou/rename-or-lesser-range-property-hint

akien-mga committed:
Merge pull request #65260 from Calinou/doc-audio-input-permissions

akien-mga committed:
Merge pull request #65248 from Mickeon/docs-improve-icons-non-breaking-space

akien-mga committed:
Merge pull request #65264 from MewPurPur/all-hail-icon-annotation

akien-mga committed:
Merge pull request #65256 from RedMser/unrounds-your-corners

akien-mga committed:
Merge pull request #65181 from fire-forge/corner-details

akien-mga committed:
Merge pull request #65250 from YuriSizov/theme-owner-rises

Calinou committed:
Use verbose prints when registering cameras in CameraServer

This prevents lines from being printed every time the editor or project
starts in the editor Output log.

akien-mga committed:
Merge pull request #65267 from Calinou/cameraserver-print-verbose

TokageItLab committed:
Fix rest fixer for the model has skin which non inverse global rest

Rindbee committed:
Fix Skeleton3DEditor crash

akien-mga committed:
Merge pull request #65279 from Rindbee/fix-Skeleton3DEditor-crash

Fix Skeleton3DEditor crash

bruvzg committed:
Re-enable per-pixel transparency support on Linux, macOS, and Windows (for Vulkan and OpenGL rendering drivers).

Faless committed:
Fix arm32 detection in Engine.

The `__ARM_ARCH_7A__` and `__ARM_ARCH_7S__` are not enough, since they
do not cover e.g. `__ARM_ARCH_8A__` in 32 bit mode, so instead of trying
to cover any possible ARM version that can work in 32 bit mode, we
can replace it with the `__arm__` macro which is defined for arm32 only
(arm64 uses `__aarch64__`).

YuriSizov committed:
Un-defer the initial theme changed notification in Window

MJacred committed:
Document Vector4i

akien-mga committed:
Merge pull request #63026 from Xwdit/improve_scene_tree_timer

Some improvements to the SceneTreeTimer

akien-mga committed:
Merge pull request #65285 from YuriSizov/window-theme-change-is-imminent

Un-defer the initial theme changed notification in `Window`

akien-mga committed:
Merge pull request #65284 from MJacred/docs/vector4i

Document Vector4i

AlmightyLaxz committed:
Stop ShapeCast3D drawing when debug collisions off

paddy-exe committed:
Disable Output port preview of Linear Depth  Visual Shader node

This disables the output port preview of the node because it's enabled preview caused a shader compilation error

raulsntos committed:
C#: Connect only once for each signal of a script

Since the list of signals in `CSharpScript::event_signals` retrieved
from calling `ScriptManagerBridge.UpdateScriptClassInfo` already
includes the signals from base scripts there is no need to iterate the
hierarchy again on `CSharpInstance::connect_event_signals`.

akien-mga committed:
Merge pull request #65111 from MatthewZelriche/DroppedXEventFix

Fix dropped XEvents early in main window lifetime.

neikeq committed:
Merge pull request #65259 from AlmightyLaxz/dotnet-build_assemblies-double-support

Add float arg to build_assemblies.py

KoBeWi committed:
Remove pathed get_node() use from scene tree dock

neikeq committed:
Merge pull request #65296 from raulsntos/dotnet/signal-once

C#: Connect only once for each signal of a script

akien-mga committed:
Merge pull request #65283 from bruvzg/reenable_per_pix_transp

Re-enable per-pixel transparency support on Linux, macOS, and Windows.

akien-mga committed:
Merge pull request #65301 from KoBeWi/get_serious()

Remove pathed `get_node()` use from scene tree dock

Rindbee committed:
Fix Skeleton3DEditor causing crash in sanitizer enabled builds

The life cycle of the `Skeleton3DEditor` singleton is too short, and when it is destroyed,
part of the logic needs to be called when its child nodes exist.

rburing committed:
Create GDExtension classes for PhysicsServer2D

This allows a 2D physics server created entirely from GDExtension.
Based on the structure of PhysicsServer3DExtension by reduz.

timothyqiu committed:
Fix crash when encoding freed object in ConfigFile

timothyqiu committed:
Fix crash when executing `LightmapGIData._set_light_textures_data`

akien-mga committed:
Merge pull request #65024 from yedpodtrzitko/yed/fix-macos-missing-vulkan

fix: graceful exit on macOS with missing Vulkan SDK

clayjohn committed:
Merge pull request #65326 from timothyqiu/element-check

Fix crash when executing `LightmapGIData._set_light_textures_data`

akien-mga committed:
Merge pull request #65299 from paddy-exe/linear-depth-disable-output-preview

Disable Output port preview of Visual Shader nodes using SpatialShader specific built-ins

aaronfranke committed:
Don't try to read values from null cameras and lights in GLTF

MatthewZelriche committed:
Fix XSendEvent crash & bootsplash.

Fixes a crash due to an Xlib error, as well as ensures that Godot
holds the correct size of the window after window modes have been
applied, before exiting the DisplayServerX11 constructor. This ensures
the bootsplash will be displayed with the correct dimensions.

Fixes #65320

aaronfranke committed:
Fix some bugs with Vector4 in C#

timothyqiu committed:
Fix crash when executing `CSGMesh3D.set_mesh` with headless Godot

aaronfranke committed:
Minor fixes to Vector4

akien-mga committed:
Merge pull request #65343 from MatthewZelriche/XSendEventCrashFix

Fix XSendEvent crash & bootsplash.

akien-mga committed:
Merge pull request #65327 from aaronfranke/gltf-null

Don't try to read values from null cameras and lights in GLTF

akien-mga committed:
Merge pull request #65278 from TokageItLab/fix-node-transform-skin

Fix `RestFixer` for the model has `Skin` which non-inverse-global-rest

akien-mga committed:
Merge pull request #65346 from aaronfranke/cs-fix-vec4

Fix some bugs with Vector4 in C#

akien-mga committed:
Merge pull request #65266 from raulsntos/dotnet/reload-non-tool-scripts

Create script instance of reloaded scripts even if they're not tools

akien-mga committed:
Merge pull request #65348 from aaronfranke/fix-vec4

Minor fixes to Vector4 in core

akien-mga committed:
Merge pull request #65347 from timothyqiu/dummy-skin

Fix crash when executing `CSGMesh3D.set_mesh` with headless Godot

akien-mga committed:
Merge pull request #65323 from timothyqiu/variant-wild

Fix crash when encoding freed object in `ConfigFile`

akien-mga committed:
Merge pull request #65028 from groud/fix_split_container

Improve SplitContainer behavior, keeping asked split position whenever possible

akien-mga committed:
Merge pull request #64840 from maximkulkin/graph-edit-node-selection

Add GraphNode 'selected' and 'deselected' signals, simplify GraphEdit

SaracenOne committed:
Freeze line when releasing mouse to create a new transition.

kisg committed:
Fix UTF-8 validation in static checks

Use isutf8 instead of recode to detect invalid UTF-8 sequences.

Also add the necessary dependencies to run the static checks locally
using act (https://github.com/nektos/act) with the Medium size image.

SaracenOne committed:
Fix UndoRedo when creating new AnimationStateMachine transition and node.

akien-mga committed:
Merge pull request #65140 from Mickeon/editor-spin-slider-newline

akien-mga committed:
Merge pull request #63568 from santagada/patch-1

akien-mga committed:
Merge pull request #65321 from rburing/physics_server_2d_extension

Create GDExtension classes for PhysicsServer2D

akien-mga committed:
Merge pull request #65356 from V-Sekai/state_machine_undo_redo_fix

akien-mga committed:
Merge pull request #65306 from Rindbee/fix-Skeleton3DEditor-causing-crash-in-sanitizer-enabled-builds

akien-mga committed:
Merge pull request #65292 from AlmightyLaxz/prevent-shapecast3d-debug-without-debugcollisions

akien-mga committed:
Merge pull request #65358 from migeran/fix-utf8-checks

akien-mga committed:
Merge pull request #65282 from Faless/fix/4.x_arm32_detect

Jummit committed:
Optimize and fix the scene unique name icon

This also makes it display correctly when using a light theme.

akien-mga committed:
Merge pull request #65362 from Jummit/fix-scene-unique-icon

YuriSizov commented on editor/editor_themes.cpp:
```diff
@@ -233,6 +233,14 @@ void editor_register_and_generate_icons(Ref<Theme> p_theme, bool p_dark_theme =
+		exceptions.insert("GuiEllipsis");
+		exceptions.insert("CodeFoldedRightArrow");
+		exceptions.insert("CodeFoldDownArrow");
+		exceptions.insert("TextEditorPlay");
```
You never used this icon in this version.

YuriSizov commented:
Seems fine other than that note and some issue that failed the static check. Needs to be rebased and fixed.

KoBeWi committed:
Improve the description of follow_viewport_enabled

YuriSizov committed:
Merge pull request #65230 from fire-forge/itemlist-spacing

YuriSizov committed:
Merge pull request #65178 from fire-forge/editor-fixes

Various editor theme fixes (bottom panel corner radius and scene tab bar height)

Chaosus committed:
Add compatibility class for `VisualShaderNodeFloatUniform`

m4gr3d committed:
Disable logic that triggers automatic focus gain as the Godot Editor is loaded

For the Godot Android Editor, this is an inconvenience as it causes the soft keyboard to show and block half of the view

cybereality committed:
Updated gamecontrollerdb.txt to latest to include new mappings for various controllers, including Xbox One, Nintendo Switch, Steam Deck, and 8BitDo

akien-mga committed:
Merge pull request #65399 from Chaosus/vs_fix

Add compatibility class for `VisualShaderNodeFloatUniform`

akien-mga committed:
Merge pull request #65371 from KoBeWi/pseudolayers

Improve the description of follow_viewport_enabled

akien-mga committed:
Merge pull request #65401 from cybereality/updated-gamepad-mapping

akien-mga committed:
Merge pull request #65354 from V-Sekai/state_machine_freeze_new_transition

Mickeon committed:
Fix tool script button color overriding custom type color

Also adds a small note on the tooltip noting the custom type.

akien-mga committed:
Merge pull request #65402 from m4gr3d/disable_android_editor_auto_focus_main

akien-mga committed:
Fix various uninitialized member pointers

Using this command:
```
find -name "thirdparty" -prune -o -name "*.h" -exec sed -i {} -e '/return /! s/\t\([A-Za-z0-9_]* \*[A-Za-z0-9_]*\)\;/\t\1 = nullptr;/g' \;
```

And then reviewing the changes manually to discard the ones that don't
seem correct/safe/good (notably changes to `core` unions).

akien-mga committed:
Merge pull request #65207 from RedMser/annotation-search

Allow searching for annotations in help

akien-mga committed:
Merge pull request #62061 from fire-forge/multinodeedit

Improvements to multi-node editing

akien-mga committed:
Merge pull request #65413 from Mickeon/fix-scene-tree-script-grayed-icon

Fix tool script button color overriding custom type color

akien-mga committed:
Merge pull request #65415 from akien-mga/fix-uninitialized-member-pointers

Fix various uninitialized member pointers

volokh0x committed:
Fix typos in Dictionary.xml

akien-mga committed:
Merge pull request #60115 from V-Sekai/scene_scale

YuriSizov committed:
Merge pull request #65417 from volokh0x/to-fix-typo-dictionary

Fix typos in Dictionary.xml

YuriSizov commented:
I think you've rebased it against the master.

PhoenixFlame101 commented:
> I think you've rebased it against the `master`, when you should've used `3.x`.
> 
> You probably need to open a new PR at this point.

Yeah my bad, working on it 😅

PhoenixFlame101 commented:
I seem to have fixed the issue; rebased with 3.x and cherry-picked the commit.

Is it better to open a new PR with the same name or reopen this one? The changes I made just now aren't reflected here because the PR closed I think.

The static errors were because I forgot to run svgcleaner on two of the new icons.

KoBeWi commented:
It's impossible to reopen a PR if the branch was force-pushed while it was closed. (and you can't change the branch)

YuriSizov commented:
@PhoenixFlame101 Yeah, open a new one please.
